File: DSA/AYUSHMEHTA_590015131/day1question2.cpp
// Time Complexity O(n) and space O(n)
#include <iostream>
#include <vector>
#include <unordered_map>
using namespace std;

void findMissingAndRepeated(const vector<int>& arr) {
    unordered_map<int, int> freq;
    int n = arr.size();

    for (int num : arr) {
        freq[num]++;
    }

    int missing = -1, repeated = -1;

    for (int i = 1; i <= n; i++) {
        if (freq[i] == 0) {
            missing = i;
        } else if (freq[i] == 2) {
            repeated = i;
        }
    }

    cout << "repeated: " << repeated << endl;
    cout << "missing: " << missing << endl;
}

int main() {
    int n;
    cout << "Enter the number of elements: ";
    cin >> n;

    vector<int> arr(n);
    cout << "Enter the array elements: ";
    for (int i = 0; i < n; i++) cin >> arr[i];

    findMissingAndRepeated(arr);


}

File: DSA/Angel_500119630/Day_1__Question_2.java
public class Day_1_Question_2{
    public static void main(String[] args) {
            int num[] = {3, 1, 3}; 
            int n = num.length;
            int result[] = missAndRepeated(num, n);

            System.out.println("Missing Number: " + result[0]);
            System.out.println("Repeated Number: " + result[1]);
    }
    public static int [] missAndRepeated(int[] nums, int n) {
        int[] freq = new int[n + 1];
        int missing = -1;
        int repeated = -1;

        for (int i = 0; i < n; i++) {
            freq[nums[i]]++;
        }
        for(int i =1 ; i <= n; i++) {
            if (freq[i] == 0) {
                missing = i;
            } else if (freq[i] > 1) {
                repeated = i;
            }
        }
      return new int[] {missing,repeated};
}

}
